### ISSUE TITLE
UI improvements for Firefox add-on running on Android

### ISSUE THREAD

**Issue by shytikov** [gorhill/uBlock#1890]
### Describe the issue

This is cosmetic issue I'm using uBlock Origin on Android Firefox. I don't know, is it possible come up with some adjustments on how uBlock elements look like on mobile, but it would be nice to have them optimized for mobile use. 

Now I have FullHD screen and I can live with it, but previously with smaller screens it was unbelievably painful. 

Note huge white space areas, and some missing descriptions for a checkboxes. Also on status popup you can see elements that could not be used for mobile phone, like eyedropper. Ok, they could be used... probably, but due to small screen it would be painful. So on my opinion they should not be displayed at all.

I don't know is it supported, but it would be nice to have add-on determine, is it running on mobile or desktop and provide user corresponding UI.
### Screenshot in which the issue can be seen

![photo815675162538125380](https://cloud.githubusercontent.com/assets/1185905/17616735/5e7a2a3e-6076-11e6-9356-c912a802111e.jpg)

![photo815675162538125379](https://cloud.githubusercontent.com/assets/1185905/17616734/5e76d500-6076-11e6-80d1-25103028fe3d.jpg)
### Steps for anyone to reproduce the issue
1. Install Firefox on Android phone;
2. Install uBlock Origin to Firefox;
3. Open uBlock Origin settings
4. Open uBlock Origin popup showing number of adds blocked.
### Your settings
- Browser/version: Firefox 48 for Android / MIUI 7 over Android 6 / Xiaomi Mi4
- uBlock Origin version: 1.8.4

**Comment by klaasb** [comment]
It's even harder with the advanced mode, where you can use the different colors to allow/passthrough/block certain other domains/objects. After using it for quite some time now, I'm able to reliably hit the "grey" that I use to allow images/etc. from certain other domains, but still not always.

uBlock Origin in general really works very well on Firefox on Android, but the UI is a bit difficult to use with touch.

**Comment by shytikov** [comment]
I think `Firefox` + `uBlock Origin` is the only way to get distraction free browsing experience on Android.  All other browsers either not support add-ons, or shipped with none-customizable blocker, which is not a option for many cases.

**Comment by jjohns71** [comment]
I would agree, a specific UI for mobile would be awesome. I know for example, just to be able to select the element picker on Firefox for Android I have to zoom in like 500x into the small area where the dropper icon is to get a finger touch to register on that particular element. There is definitely a lot of "white space" that isn't really optimized for a small mobile screen.

@gorhill How could the community help implement & test a better mobile UI?

**Comment by kazzkiq** [comment]
I've got so excited about uBlock, and it run so much better than ABP that I got really into helping making this tool work better for mobile (and desktop) devices. Here is a (tiny tiny) sneak peek of what I've been thinking for it:

_**(Some labels are not in english because I only had a portuguese version of uBlock running on my machine)**_

![1234567](https://cloud.githubusercontent.com/assets/1953194/17641095/d6b02806-60e8-11e6-9079-030fdc2ad4f6.jpg)


Also, a desktop version:

![configuration](https://cloud.githubusercontent.com/assets/1953194/17641105/393944b2-60e9-11e6-90a5-75ae3453cf3e.png)


I would create a new Issue to propose some UI improvements, but just noticed that @gorhill do not accept *"design proposals"* as Issues, so I'm really not sure if the project is open to this kind of proposals at all. But anyways, if there is some space to work on the code to make this happen, I would gladly help build it (most of the current DOM structure and JS plugins remain as the same, so it should not take much effort to make those visual changes).

**Comment by gorhill** [comment]
@kazzkiq I like what I am seeing in your pics. I am willing to bring in your design -- as long as for now the redesign is limited to the dashboard. Is this really just CSS work?

**Comment by kazzkiq** [comment]
@gorhill Yep. I would say 4/5 CSS work, 1/5 HTML or JS changes.

As the core behaviors (tabs/views, collapse-panel, etc) already work pretty well, I think the work would be put mostly into styling the dashboard and adding some classes to some elements. Small HTML rearrangement would be needed to make a prettier mobile menu, but analyzing the Dashboard code, I can't see any other considerable HTML change besides the mobile menu.

If you want to go forward with this I can show some other screens from the dashboard and start implementing it this week.

**Comment by klaasb** [comment]
For me, the most important UI improvements for mobile (which the original issue was about) could be done to the panel in the second screenshot in the original post and especially the advanced view of that:

![screenshot_20160813-092405](https://cloud.githubusercontent.com/assets/961344/17642316/3bb084d6-6138-11e6-8b49-1208a50d3c82.png)

This is really hard to use by touch on a 4.7" 1080p phone screen - especially selecting red/grey/green.
I don't need to access the dashboard on mobile very often/at all.

**Comment by Hrxn** [comment]
Speaking of the dashboard (and UI/UX improvements):

One thing that always felt a bit strange somehow and not really intuitive, is that the default dashboard tab (i.e. where you always land when accessing the dashboard) is the settings pane. Because in normal usage, normal browsing etc. it is never my actual destination. I open the dashboard and switch to 'My Filters' or '3rd Party Filters' immediately. I basically use the settings tab only once, to pick my preferences, and then I'd never go there again. I don't know what you guys are doing there or how you are using the extension, but I don't really get the thought process behind this. Maybe it's just me..

So, I would propose to either...
- A: The no work involved solution: Just rearrange the tabs in the dashboard.
- B: The work involved solution: Use a simple overview/status page as the default dashboard pane
(This could basically just show the same stats as displayed right now on different panes, for example) 

I don't know. Just my two cents..

**Comment by jjohns71** [comment]
@Hrxn just to add a little to your discussion point, the only thing I use on the landing page frequently is the 'Backup to file' option. Honestly though, this could be moved to the 'About' tab to mirror uMatrix functionality and I would have no problem with that. I agree that most of the other Settings are probably set once and then never changed for the average user. (That is the case for me at least).

It does seem to me that the majority of users would find the '3rd-party filters' or 'My filters' tab more useful as the default landing page.

One additional thought if we are thinking of UI improvements for uBlock, does it make sense to include a link to the logger on one of the landing tabs? Not sure if this would change the resources needed to run the extension but another idea is to just incorporate the logger as a tab from the dashboard. From monitoring the issues submitted here over the last few months I think a lot of people who use uBlock have no idea the logger even exists because it is hidden behind a graphic icon that doesn't even have "alt text" when you hover over it.

**Comment by spectatorBH** [comment]
Let me mention a bit of my ideas, too:

1. @Hrxn  I agree the initial default tab might be changed or even created a new special one, but I would also suggest to remember the last open dashboard tab (also between browser sessions?), so it would be easier to switch back and forth while tweaking some page;

2. remember pages/lists scroll positions when swithing tabs until the dashboard is closed -- imho, especially critical for My Filters list and to a lesser extent for 3d Party Filters tab when troubleshooting/experimenting/setting up;

3. add icons or buttons for quick jump to the top or bottom of the current page or list" -- for more convenient navigation especially on mobile screens;

4. @klaasb advanced view of popup panel on mobiles should be split into 2 parts (now left and right) as either 2 distinct tabs or windows, to accomodate for limited horizontal space (maybe only if current horizontal resolution is less than some predefined const);

5. @klaasb color coded regions of the advanced view popup (at least on mobiles) should continiously cycle-switch their state when touched/clicked (green->dark-grey->red->default->green->...) -- this would effectively triple the active width of these controls and make it easy to operate them with fingers not only on Android, but also on touch-enabled Windows tablets/hybrids;

6. @jjohns71 if a logger tab proposal is ever implemented (nice idea!) I guess it should open in 2 different modes: a) instantly active (enabled) when open via direct icon on the popup panel and b) passive (disabled) when any other dashboard tab was meant to be opened first. Later on throughout the session it could be started/stopped (enabled/disabled) manually via some dedicated control on its tab. This way the logger would be both more accessable/visible and not suck any resources when not exactly needed.

P.S. @kazzkiq I'm quite impressed with your initial sketch design, please show us other parts! Also, do you have any ideas for popup panel UI improvements?

**Comment by gorhill** [comment]
To be clear: I am opened for a re-skin of the dashboard for now. To keep a possible re-vamp of the dashboard as simple as possible, I prefer everybody hold onto all their ideas of what everything should look like, that is an endless debate and will drown out what the narrow "reskin dashboard" issue.

**Comment by kazzkiq** [comment]
@spectatorBH Here are other screen samples:

![12345](https://cloud.githubusercontent.com/assets/1953194/17676639/e309e834-6305-11e6-976c-1911cb20d5ad.jpg)

I agree with @gorhill. Perhaps we should start only with dashboard then in the future, once its finished, and once people approves the new UI version, the maybe we look forward to change other parts with the project owner's approval. For now, dashboard is a good (and ambitious) challenge.

@gorhill I can make a fork and start working on the code already, then once its stable enough I can send a PR for your approval. Its that ok for you?

**Comment by gorhill** [comment]
In 3rd-party filter lists, I like it better if the visual emphasis is more on the checkbox and the name of the list (I still prefer a checkbox than a switch-like UI). The visual for the purge-cache and out-of-date buttons need to be less prominent -- these are secondary information, the primary one being the list name and whether it is selected.

**Comment by FatOrangutan** [comment]
Is the dashboard redesign still on the cards? I ask because I like it a lot and because a fork or any other discussion hasn't happened for a little while now.

**Comment by UserNaem** [comment]
@kazzkiq Please don't use those switch-like checkboxes. They are not "modern", they are just bad. Normal checkboxes serve the same purpose while occupying less space, and are not any harder to hit with fingers either.

**Comment by Hrxn** [comment]
Visual recognizability > Occupying Space

**Comment by UserNaem** [comment]
Yes, they are also ugly. And side vision won't tell you their current state, you have to actually look at them to see their state.

**Comment by FatOrangutan** [comment]
As proof of concept, decided to make some quick and dirty CSS only changes to the dashboard. I made some progress on the first two panes:

![chrome_ublock0_content_dashboard html settings html155435](https://cloud.githubusercontent.com/assets/8336638/18508909/df963d4c-7a93-11e6-92cd-c0a231010061.png)
![chrome_ublock0_content_dashboard html 3p-filters html155357](https://cloud.githubusercontent.com/assets/8336638/18508916/e5cc4616-7a93-11e6-87ca-d802ba9903b0.png)
In action: https://u.pomf.is/irapyz.gif (~3MB gif)

**Comment by Hrxn** [comment]
> Yes, they are also ugly.

100% subjective.

> Side vision

?
Not sure what you mean here, but, increased size of the UI element, plus use of color, plus use of increased contrast is definitely better for vision than tiny greyish-on-greyish checkboxes.

**Comment by SW1FT** [comment]
@FatOrangutan Good job on that PoC, could you make a repository for it?

**Comment by UserNaem** [comment]
@FatOrangutan Good stuff, please post the CSS.
@Hrxn 
> 100% subjective.
Indeed.
> tiny greyish-on-greyish checkboxes
Normal checkboxes were made with mouse and keyboard in mind, whereas "slider" checkboxes were made for touch devices to prevent the user from accidentally switching them. With mouse that's never been a problem.

**Comment by gwarser** [comment]
>whereas "slider" checkboxes were made for touch devices to prevent the user from accidentally switching them

How?

**Comment by UserNaem** [comment]
If you try to swipe them in the wrong direction they won't work. Not all touch applications behave like that, though.

**Comment by kazzkiq** [comment]
Sorry folks, I've been deeply engaged in a project these weeks and couldn't start working on the implementation until now.

@gorhill I've made little changes to the 3rd-party filter lists with more focus on the names instead of the secondary information:

<img width="633" alt="captura de tela 2016-09-14 as 16 15 04" src="https://cloud.githubusercontent.com/assets/1953194/18525890/cebf8ae0-7a94-11e6-8249-48adfe79ef8e.png">


### Checkboxes vs Switchers

While I agree that for desktops a checkbox might be enough (and maybe even better for UX, as users are already used to it), on mobile-devices switches are way more used. This is better for user recognition, and easier to deal with a finger than a mainstream checkbox (better usability).

So I would say the best option would to keep checkboxes in desktop devices, while showing switches to mobile users. Note that this also can be easily achievable with only CSS (and maybe a little change in the checkboxes HTML).

![12345](https://cloud.githubusercontent.com/assets/1953194/18526252/2d487530-7a96-11e6-8d79-0a3512615fc7.png)


@FatOrangutan Neat! Could you please make the code available so we can also help implementing it? Perhaps we should define an oficial repo-branch to start working/testing the new design?

**Comment by FatOrangutan** [comment]
I have attached the CSS code / userstyle. I worked a bit for the remaining pages in the dashboard and made some slight improvements (fixed left border vanishing for a table etc.) to the ones in my previous comment. Here's how it looks like now: http://imgur.com/a/oGmox  
  
Please bear in mind that I'm new to CSS and the code is very messy. It was also originally meant just for my personal use so it wouldn't scale very well at all. I am on 1366x768, and it should look mostly fine at this resolution and maybe even resolutions close to it. I was using this with Stylish, so accordingly the `!important` declarative has been freely abused. The code for some of the things - like making `#externalListsDiv` seem to be within the `#lists` element on the 3rd Party Filters page - are outright hacky but doing that cleanly requires changes to the HTML. Lastly, to make the styling easier, I kept the code for each of the pages completely separated. I realise that some of this code is common across all the pages. 

@SW1FT , @kazzkiq : I don't know much about how to do things at Github. I think it would be best if one of you handle the repo. Once that is made, we can probably also take this discussion there.

[Dashboard Reskin.txt](https://github.com/gorhill/uBlock/files/475325/Dashboard.Reskin.txt)

**Comment by gorhill** [comment]
> So I would say the best option would to keep checkboxes in desktop devices, while showing switches to mobile users

I personally like the Fontawesome switches, I used them in uMatrix. I wouldn't mind if these were used for both desktop and mobile. See:
- http://fontawesome.io/icon/toggle-off/
- http://fontawesome.io/icon/toggle-on/

There is a good enough difference between on/off, so that it makes it easy to see at a glance what is selected or not.

**Comment by UserNaem** [comment]
@gorhill At least please consider leaving an option to use classic checkboxes, not everyone is fond of them. I personally feel like they were made for the tech illiterate who would be confused when looking at an empty checkbox.

**Comment by SW1FT** [comment]
@FatOrangutan First of all, thanks for sharing the code, I really like the design. 

I did find some issues with the scaling of `#externalListsDiv` (at 3p-filters.html) mostly because I feel like it looks too small and `max-width` should scale with the `body` width. Also if you change the height enough, it will go out of the background which is something that doesn't happen on the whitelist.html and 1p-filters.html (probably requires some html fixing like you said). I also noticed that you didn't re-skin `#cloudWidget` yet which is where is shown the cloud backup data information, but that can be worked on later of course. Great job so far.

Regarding who should make the repo, it should probably be done by the person who is gonna do most of the work on this, or just @gorhill as he has the final say on everything.

**Comment by kazzkiq** [comment]
@SW1FT @FatOrangutan @gorhill 

I took the liberty to fork the project and start implementing the new design. You can check the status (and build it locally) here:

**https://github.com/kazzkiq/uBlock/**

I've added SASS as a CSS pre-processor to make DRY a reality in CSS (and help with the development speed). While this don't affect the final CSS, the build process **do** get affected a little bit. So if you wanna build the project by yourself while the new UI is on development, you may have to follow this small guide:

https://github.com/kazzkiq/uBlock/blob/master/BUILDING-NEW-UI.md

P.s. Please keep in mind this fork is in DEVELOPMENT status and may end up breaking UI and/or functionalities from uBlock Origin until finished. This mean you **should not** build an extension from this fork and use it "for production" in your browser. Things will break. Use only for testing/developing purposes. You've been advised. 🚨

**Comment by geokis** [comment]
Hey guys, first thank you @kazzkiq for your work on the UI.
I am not sure how the other user are thinking about this, but for me it is uncomfortable how the secondary information are placed.
In my opinion it is confusing that the placement is related to the size of the filter name.
So my suggestion is to bring a little bit more calmness in it.

**E.g.:**

<img width="633" alt="cebf8ae0-7a94-11e6-8249-48adfe79ef8e" src="https://cloud.githubusercontent.com/assets/7952956/18822612/206be656-83b2-11e6-819f-4c28d0710693.png">

**Comment by SW1FT** [comment]
@geokis I think it would be better if the "purge cache" icon was before the name of the filter, after the checkbox.

**Comment by Rikk** [comment]
I would like smaller paddings/margins when it is not being used in mobile. It's just not nice to use Metro-inspired ui with mouse.

**Comment by kazzkiq** [comment]
For professional reasons, I'm getting trouble to devote some time to work on this UI. It got stuck around 50% on my fork. But just to make it clear: My plan is to still deliver a usable interface to my fork until the end of December. From then on the project owner can decide to handle/merge the new UI.

If anyone is interested, [here is the fork](https://github.com/kazzkiq/uBlock) (instructions in the [BUILDING-NEW-UI.md file](https://github.com/kazzkiq/uBlock/blob/master/BUILDING-NEW-UI.md)).

**Comment by kazzkiq** [comment]
I think the fork reached a level of maturity that enables it being tested by others. 

Here are some samples already running from uBlock extension itself:

**Mobile**:
![ublock-mobile](https://cloud.githubusercontent.com/assets/1953194/21332965/4be0de0e-c633-11e6-985c-35d8ccdc4b41.gif)


**Desktop:**
![ublock-desktop](https://cloud.githubusercontent.com/assets/1953194/21332977/62ca6658-c633-11e6-87d9-97a15867c19f.gif)


It would be interesting if anyone else could help checking for any UI errors  on other browsers/devices too.

Building this fork should not take more than 5 minutes and [a guide can be found here](https://github.com/kazzkiq/uBlock/blob/master/BUILDING-NEW-UI.md).

**Comment by FatOrangutan** [comment]
A tiny issue, visible in your gif for the desktop too: on the My rules / dynamic filtering rules page, clicking 'Edit' makes it so that the rows in the right (temporary) column no longer align correctly with the rows in the left (permanent) column. You probably need to target `#diff textarea` separately.

**Comment by kazzkiq** [comment]
@FatOrangutan nice catch. Already fixed.

**Comment by FatOrangutan** [comment]
One more: on the 3rd-party filters page, the last section ('Custom') has its header within the main area above but its instruction line, its text box and the button that follows it are outside it. Here's a screenshot:
![frame76](https://cloud.githubusercontent.com/assets/8336638/21337078/2f693c3c-c690-11e6-9189-49e353de57b5.png).

BTW, you had nice '!', '?' icons in your mockups that seem to not have made it into the fork. Is that intentional?

**Comment by kazzkiq** [comment]
@FatOrangutan To put the bottom part (textarea, etc) inside the "Custom" section would require some structural and programmatically changes (as the whole section is generated dynamically by JavaScript). I believe it would fall off the "CSS and HTML only" idea behind this UI change.

**BTW, you had nice '!', '?' icons in your mockups that seem to not have made it into the fork. Is that intentional?**
Those are still being being finished 😄 , will be adding soon.

**Comment by FatOrangutan** [comment]
It looks like you missed the edits in my message (maybe you are replying through email?). Here is a direct link to the message: https://github.com/gorhill/uBlock/issues/1890#issuecomment-268143466

**Comment by kazzkiq** [comment]
I had the page open by some time and didn't catch the edits 😄 .

About the first two edits: Will be taking a look at them after I wake up. They look pretty simple to fix.

**"BTW, I am mostly going by your gif since I can't check myself - I had a pretty hard time trying to use npm the last time I tried it. It would be great if you could release a compiled build."**

Here is the Chrome build I'm using at my machine:  [uBlock0.chromium.zip](https://github.com/gorhill/uBlock/files/662752/uBlock0.chromium.zip)

Will be trying to generate it to other browsers as well if other people find it useful.

**Comment by SW1FT** [comment]
@kazzkiq Can you build it for Firefox please? Thanks!

**Comment by DrunkenSasquatch** [comment]
I ran `gulp build:firefox` and recieved the following compilation error `'tools' is not recognized as an internal command`

What am I missing here? 
I already have node.js and installed gulp globally.....

**Comment by MatyCZ** [comment]
**On page 3rd-party filters:**
Move cursor to the Trash icon > Cursor is not changed to Hand

**Comment by MatyCZ** [comment]
* Selected item (page) in the menubar is really badly visible - only small underline
* uBlock logo looks like active item in the menubar with this light background

Maybe that light backgroud will be better for current active item in the menubar.

**Comment by MatyCZ** [comment]
Menubar item without `font-size: 1.3rem` looks better

```css
#dashboard-nav-widgets .menuTab .tabButton
```

**Comment by kazzkiq** [comment]
@SW1FT Firefox version:  [uBlock0.firefox.zip](https://github.com/gorhill/uBlock/files/664066/uBlock0.firefox.zip)

@MatyCZ I've made some improvements on your points. You can generate them locally to check the updates (will be generating the extensions again at the end of the day).

**Comment by DrunkenSasquatch** [comment]
The `apply changes` button and the check boxes appear different in Firefox. Inspect element shows 
`<button id="buttonApply"`

Screenshot:
![argarf](https://cloud.githubusercontent.com/assets/16097892/21359875/5b45392a-c729-11e6-8f2f-3020032cd540.jpg)

**Comment by SW1FT** [comment]
I think `font-size: 1.5rem;` in `#dashboard-nav-widgets .menuTab .tabButton` looks better than `font-size: 1.3rem;`. Also, will you restyle `#cloudWidget`?

**Comment by gwarser** [comment]
Barely visible textbox

![2017-03-15-15-21-08](https://cloud.githubusercontent.com/assets/886325/23991993/91ffcd18-0a3b-11e7-9f4f-723ea6aba9db.png)

**Comment by gorhill** [comment]
What's the screen size of your device?

**Comment by jjohns71** [comment]
@gorhill I have a 5.7 inch screen (2560x1440) and have the same issue. It is impossible to see or scroll to the right through the text after the first few characters in both the Permanent Rules & Temporary Rules boxes.

Screenshot below:

![screenshot_20170316-101106 2-cleaned](https://cloud.githubusercontent.com/assets/6795795/24000546/df6da4a8-0a31-11e7-90d7-d81efa90d988.jpg)

**Comment by gwarser** [comment]
4.3" 480x800 :)

**Comment by gorhill** [comment]
I can adjust CSS rules to regain a bit more space vertically, but ultimately there is no magic trick to turn a small display into a larger one.

**Comment by gwarser** [comment]
Much better.
Cannot this be folded to single "hamburger" button?

**Comment by krzysztof-sikorski** [comment]
I think it could be better to not display "permanent rules" & "temporary rules" side-to-side, but instead display them one below the other, with visible margins around textareas to ease vertical scrolling.

That way it would be easier to edit rules, and the only disadvantage I can think of is that it would be no longer possible to compare both panels with one glance - but the "temporary rules" panel marks visually what was added/deleted and what not, so I think it is not as big issue as cumbersome editing&scrolling.

**Comment by gorhill** [comment]
> I think it could be better to not display "permanent rules" & "temporary rules" side-to-side, but instead display them one below the other,

This needs redesign. Looking at it again, I should eliminate the redundancy of showing twice the same rules. One line per rule with a padlock + remove icons for each rule should be sufficient. The _padlock_ tells whether the rule is in the permanent ruleset, the _remove_ icon allows to remove the rule from the temporary ruleset.

**Comment by nikrolls** [comment]
@gwarser Bear in mind that UX designers are moving away from hamburger menus wherever possible because of the discoverability issues they create, not to mention they're very fiddly to open on large screens. It's better to have your menu in plain sight.

**Comment by Serkan-devel** [comment]
@kazzkiq might the new colorful UI impact performance? My only HTML knowledge comes from an old HTML 3.2 book that was randomly in a bookshelf in my room.
@gorhill when will kazz's design be available in a stable ublock release?

**Comment by Serkan-devel** [comment]
bump

**Comment by bershan2** [comment]
@kazzkiq Great work!

First of all, **when do you plan to merge the main repository changes into your repository?** Unfortunately, your repo is a bit behind the main repo (by 471 commits at time of writing), but automatic merging resolves almost all differences. However, problematic files include `src/whitelist.html`, `src/settings.html`, `src/dashboard.html`, and `src/3p-filters.html`  (as well as almost all `src/css/*` which, of course, was expected). Most differences are pretty minimal (e.g. indentation) and fixing them would take 15 min at most, but they would steel need to be resolved and tested before merging in with the main repository. When do you plan on doing that? If I resolved these issues and submitted a pull request would you accept it? (I'd like to test the new design with the contemporary uBlock Origin version.)

Secondly, **may I use your style sheets for a website?** Probably, "yes", because it is under GPLv3, but wanted to double-check.
I was ~~thinking~~ dreaming of porting the uBlock Origin Wiki to a Markdown (probably GFMD) site with address like `gorhill.github.io/uBlock` or `uBlockOrigin.github.io`. That would make it more (non-programmer) user-friendly and help new users "discover" uBlock.

**Comment by bershan2** [comment]
@Serkan-devel 
> when will kazz's design be available in a stable ublock release?

As you can [see](https://github.com/kazzkiq/uBlock/commits/master), @kazzkiq did all this in a bit over a month 16 Sep to 20 Dec, 2016, and then stopped, so there is no way to set a definitive release date. Besides, this is a rather large change, so would need polishing (both visual and technical) before entering stable release.

**Comment by sabret00the** [comment]
@kazzkiq I notice one of the issues you've laid out is that you feel the iconography regarding synchronisation needs to be swapped/updated. But given that such large changes are being made in regards to the look and feel of uBO, wouldn't it make sense to change that iconography later rather than sooner in order for users to maintain familiarity .

Also I quite like what you've done with the mobile version. It's a vast improvement in terms of usability, but I noticed that there are some aspects where the information icon seems to drop to the line below the element. That doesn't seem right.

**Comment by kazzkiq** [comment]
@bershan2 Just merged. Now the repositories are synced.

> Secondly, may I use your style sheets for a website?

Of course! The code is under GPLv3 but honestly all my changes should go to the [WTFPL](http://www.wtfpl.net/). Do/copy/modify whatever you want.

I also updated some styles missing (like the cloud-storage panel).

Following @sabret00the tip, I'm not going to update all the icons. That would impact other areas of uBlock that also use them, and its out of scope right now.

The only screens that still need working are:
- Settings (mobile)
- 3rd Party Filters (mobile)

All the other pages are production-ready.

As always, instructions for building and testing locally [**can be found here**](https://github.com/kazzkiq/uBlock/blob/master/BUILDING-NEW-UI.md).

**Comment by sabret00the** [comment]
@kazzkiq did you submit a pull request? I'm not sure @gorhill is keeping an eye on this thread.

**Comment by Serkan-devel** [comment]
I think gorhill merged it by himself, I can't find the PR though
![auswahl_233](https://user-images.githubusercontent.com/16866642/29494268-b616f0a0-85a6-11e7-94f1-826c73539bac.png)

**Comment by gorhill** [comment]
> @kazzkiq did you submit a pull request? I'm not sure @gorhill is keeping an eye on this thread.

 I just looked at the result and I will not be merging the changes.

**Comment by kazzkiq** [comment]
@sabret00the there are still some minor tweaks to get to 100% before submitting a PR.

@gorhill Any specific reasons, so we can work on it?

**Comment by gorhill** [comment]
> Any specific reasons, so we can work on it?

Too much subjective changes -- some may like the changes, others will dislike the changes. Where subjectivity is involved, status quo prevails.

**Comment by DrunkenSasquatch** [comment]
> some may like the changes, others will dislike the changes

@gorhill I think almost everyone can agree that @kazzkiq design is better than the current. So what exactly is the problem? I cannot follow the logic on having no design whatsoever.

**Comment by sabret00the** [comment]
While we may all feel that it's a vast improvement in terms of aesthetics and perceptive usability, Raymond is the one that has built and maintained this extension, if he's not happy with it then that's his decision and we must respect that.

**Comment by kazzkiq** [comment]
What I've noticed is that the new CSS ends up "leaking" to other parts of the extension like the popup, events window, etc, perhaps those are examples of subjective changes @gorhill commented? The options page CSS shouldn't affect other parts of the extension, but is relatively simple to solve.

**Comment by Serkan-devel** [comment]
How will this actually work as web-extension?

**Comment by heubergen** [comment]
> Too much subjective changes -- some may like the changes, others will dislike the changes. Where subjectivity is involved, status quo prevails.
@gorhill I can understand the argumentation, but I have to disagree strongly. The design is something never everyone will agree but that doesn't mean you should not change it at all. Let's make an official test build and then receive the feedback from the user.

**Comment by kazzkiq** [comment]
Its up to @﻿gorhill and community. If its decided to give it a chance, development will be retaken.

**Comment by DrunkenSasquatch** [comment]
gorhill said [back in 2016](https://github.com/gorhill/uBlock/issues/1890#issuecomment-239599840) that he would "bring in your design", then 12 months later the design is too "subjective". Interesting because the original proposal is basically the same which was offered in 2016, why did he change his mind?

Following that logic, isn't every user interface design subjective to the designer? 😆

**Comment by shytikov** [comment]
Hi All,

I'm the original topic starter from 2016, I haven't been an active participant so far but latest comments made me feel worried. 

Please scroll up to my first post. I'm not complaining about design, which would be indeed very subjective. And subjectively I would definitely appreciate re-design according to most modern UX practices.

But my complaints were about **USABILITY** on mobile devices. And this is objective. On a smaller vertical screen dealing with uBlock Origin UI is not easy. The control elements just too small and their positions not very accessible!

Can we at least make an effort to make UI adoptive to user screen size? Currently, [there are more mobile internet users than desktop](https://www.stonetemple.com/mobile-vs-desktop-usage-mobile-grows-but-desktop-still-a-big-player/), so it would be nice to make uBlock Origin more friendly to smaller screens.

**Comment by gorhill** [comment]
Here is what I would like to approach this issue:

- Suggested changes committed in a github fork constantly kept up to date with current master so that this can easily be tested by everyone at any time.
- Very _focused_, _incremental_ changes which improve obvious specific usability issue.
    - No grand redesign, it has to be incremental so that I can manage the changes and to eliminate chances of regressions.
    - First focus on usability issues which affects the wide majority of users, not the fraction of percent which uses the _"My rules"_ pane for example.
- Changes to the look and feel must be _incremental_ -- uBO has to still look familiar enough at every single iteration.
- Mind repercussions of changes on all supported platforms.

**Comment by jspenguin2017** [comment]
Anyone wants to have another look at this in Nano? 
https://github.com/NanoAdblocker/NanoCore/issues/33

**Comment by Serkan-devel** [comment]
Have issues just been unlocked here?

**Comment by gwarser** [comment]
This should probably be closed now.

**Comment by gorhill** [comment]
Closing as suggested. The UI work is still ongoing but it's not done as per this specific issue (which has be stale for years now), it's done with input from Mozilla's @brampitoyo and the end goal is to make uBO's UI fits Firefox Preview's own design.
